### PR TITLE
feat: add id stitcher debugger model type

### PR DESCRIPTION
**Pull request by shekhar-rudder**
## Description of the change

This PR adds a new pynative model type `id_stitcher_debugger` in `profiles_mlcorelib` repo. Currently it implements YamlReport. More reports will be added in later PRs

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/debugger.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            return
+        id_stitcher_models = {}
+        # With the current implementation of "models" method, it shouldn't be called in "register_dependencies" method
+        # because it will try to compile itself and that will cause a deadlock
+        models = this.base_wht_project.models()
+        for model in models:
+            if model.model_type() == "id_stitcher":
+                id_stitcher_models[model.name()] = model
+        if len(id_stitcher_models) > 1:
+            selected_model_name = self.reader.get_input(
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/id_stitcher/debugger.py:58]: Is `reader` an addition to the PyNativeRecipe?

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/yamlReport.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            for i, component in enumerate(components):
+                print(i + 1, ":", component)
+            print(
+                "These",
+                len(components),
+                "groups are not connected to each other, so one id within each group is never connected to another id in a different group. This typically creates an understitched graph",
+            )
+        missing_id_types = []
+        for id_type in self.entity["IdTypes"]:
+            if id_type == "rudder_id":
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/id_stitcher/yamlReport.py:71]: Does it cover `user_main_id` etc?

**Review by dpatchigolla — APPROVED** [review]

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/debugger.py (hunk 1 above)
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/id_stitcher/debugger.py:58]: Yes

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/yamlReport.py (hunk 2 above)
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/id_stitcher/yamlReport.py:71]: Yes

**Pull request merged**